### ISSUE TITLE
*feature request* How to Print a variable ?

### ISSUE THREAD

**Issue by zxqchat** [tensorflow/tensor2tensor#753]
### Description

> … I have restore some variable in pre-trained model-ckpt files, and I need to print these variable to confirm that every variable is successed or not ?

### *TensorFlow* and *tensor2tensor* versions

> …tensorflow-gpu==1.4.0  tensor2tensor == v1.2.9

### the code is add by me to load a pre-trained model-ckpt and print the value

> …
####add by xqzhou for  load model
`  
  if FLAGS.load_model is not None:
    all_weights = {v.name.split(":")[0]:v for v in tf.trainable_variables()}
    tf.logging.info("model build load_model in %s"%FLAGS.load_model)
    restore_variables(FLAGS.load_model)
    var_list = tf.trainable_variables()
    for var in var_list:
        name = var.name.split(":")[0]
        # with tf.Session() as sess:
        #     print(sess.run(var))
        if name in all_weights:
            tf.logging.info("Restore %s for initial" %var.name)
            tf.assign(var, all_weights[name])
    with tf.Session(config=tf.ConfigProto(log_device_placement=True)) as sess:
      for (x, y) in zip(tf.global_variables(), sess.run(tf.global_variables())):
        print('\n', x, y)
`

### In case of bug report: Error log

> …InvalidArgumentError (see above for traceback): Cannot assign a device for operation 'training/gradients/transformer/body/model/parallel_1/body/decoder/layer_0/ffn/layer_prepostprocess/layer_norm/Mean_grad/Fill': Operation was explicitly assigned to /job:localhost/device:GPU:1 but available devices are [ /job:localhost/replica:0/task:0/device:CPU:0, /job:localhost/replica:0/task:0/device:GPU:0 ]. Make sure the device specification refers to a valid device.
	 [[Node: training/gradients/transformer/body/model/parallel_1/body/decoder/layer_0/ffn/layer_prepostprocess/layer_norm/Mean_grad/Fill = Fill[T=DT_INT32, _class=["loc:@transformer/body/model/parallel_1/body/decoder/layer_0/ffn/layer_prepostprocess/layer_norm/Mean"], _device="/job:localhost/device:GPU:1"](training/gradients/transformer/body/model/parallel_1/body/decoder/layer_0/ffn/layer_prepostprocess/layer_norm/Mean_grad/Shape_1, training/gradients/transformer/body/model/parallel_1/body/decoder/layer_0/ffn/layer_prepostprocess/layer_norm/Mean_grad/Fill/value)]]
